test/internal/helpers/Player.java

```diff
@@ -316,35 +801,78 @@ public static Cleanups hasRange() {
         });
   }
 
-  public static Cleanups setProperty(String key, int value) {
+  /**
+   * Sets a property for the user
+   *
+   * @param key Key of property
+   * @param value Value to set
+   * @return Restores the previous value of the property
+   */
+  public static Cleanups withProperty(String key, int value) {
     var oldValue = Preferences.getInteger(key);
     Preferences.setInteger(key, value);
     return new Cleanups(() -> Preferences.setInteger(key, oldValue));
   }
 
-  public static Cleanups setProperty(String key, String value) {
+  /**
+   * Sets a property for the user
+   *
+   * @param key Key of property
+   * @param value Value to set
+   * @return Restores the previous value of the property
+   */
+  public static Cleanups withProperty(String key, String value) {
     var oldValue = Preferences.getString(key);
     Preferences.setString(key, value);
     return new Cleanups(() -> Preferences.setString(key, oldValue));
   }
 
-  public static Cleanups setProperty(String key, Boolean value) {
+  /**
+   * Sets a property for the user
+   *
+   * @param key Key of property
+   * @param value Value to set
+   * @return Restores the previous value of the property
+   */
+  public static Cleanups withProperty(String key, Boolean value) {
     var oldValue = Preferences.getBoolean(key);
     Preferences.setBoolean(key, value);
     return new Cleanups(() -> Preferences.setBoolean(key, oldValue));
   }
 
-  public static Cleanups setQuest(QuestDatabase.Quest quest, String value) {
+  /**
+   * Sets progress for a given quest
+   *
+   * @param quest Quest to set
+   * @param value Value for quest property
+   * @return Restores previous value
+   */
+  public static Cleanups withQuestProgress(QuestDatabase.Quest quest, String value) {
     var oldValue = QuestDatabase.getQuest(quest);
     QuestDatabase.setQuest(quest, value);
     return new Cleanups(() -> QuestDatabase.setQuest(quest, oldValue));
   }
 
-  public static Cleanups setQuest(QuestDatabase.Quest quest, int step) {
-    return setQuest(quest, "step" + step);
-  }
-
-  public static Cleanups setupFakeResponse(int code, String response) {
+  /**
+   * Sets progress for a given quest
+   *
+   * @param quest Quest to set
+   * @param step Step to set
+   * @return Restores previous value
+   */
+  public static Cleanups withQuestProgress(QuestDatabase.Quest quest, int step) {
+    return withQuestProgress(quest, "step" + step);
+  }
+
+  /**
+   * Sets next response when the FakeHttpClientBuilder is active Note that for some reason setting
+   * the response seems to wipe out getRequests() at the moment
```

Comment:
It wipes out `getRequests` because it creates a new client builder which it then sets up. Also note that the fake builder doesn't have to be present before this method is called, and will be afterwards.

Reply:
Ah I see, of course. I wonder if we can give code access to the client it has

Reply:
The intent was that this function be used for only one-shot and forget calls, and ones where you want access to the client you should set it up yourself using `HttpClientWrapper.setupFakeClient()`, and then you can use `HttpClientWrapper.getRequests()`.

Reply:
We could use a more general mocking helper to do it all